Sources/SmithyJSON/Writer/Writer.swift

```diff
@@ -0,0 +1,168 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import protocol SmithyReadWrite.SmithyWriter
+import enum SmithyReadWrite.Document
+import enum SmithyTimestamps.TimestampFormat
+import struct SmithyTimestamps.TimestampFormatter
+import struct Foundation.Data
+import struct Foundation.Date
+import class Foundation.NSNumber
+
+public final class Writer: SmithyWriter {
```

Comment:
This type is used to serialize data into a tree of `Writer`s that can be used to write JSON.